src/index.js

```diff
@@ -0,0 +1,87 @@
+import './style.css';
+
+const submit = document.querySelector('.add-btn');
+const tasksDiv = document.querySelector('.tasks');
+const taskInput = document.querySelector('.add-task input');
+
+let editId;
+let isEditTask = false;
+let tasks = JSON.parse(localStorage.getItem('task-list'));
+
+window.editTask = (taskId, taskDescripton) => {
+  editId = taskId;
+  isEditTask = true;
+  taskInput.value = taskDescripton;
+};
+
+const addElemToPage = () => {
+  let div = '';
+  if (tasks) {
+    tasks.forEach((task) => {
+      div += `
+      <div class='task' data-id='task.id'>
+        <input id="${task.index}" class='checkbox' onclick='updateCompleted(this)' type="checkbox">
+        <p id='${task.index}'class='text' contenteditable="false">${task.descripton}</p>
+        <button type="button"  id="${task.index}" class='edit' >
+        <span onclick='editTask(${task.index}, "${task.descripton}")' class="material-symbols-outlined">edit</span>
+        </button>
+        <button type="button"  id="${task.index}" class='del' onclick='del(${task.index})' >
+        <span class='material-symbols-outlined del'>delete</span>
+        </button>
+        </div>
+      `;
+    });
+  }
+  tasksDiv.innerHTML = div;
+};
+addElemToPage();
+
+submit.addEventListener('click', () => {
+  const userTask = taskInput.value;
+  if (!isEditTask) {
+    if (!tasks) {
+      tasks = [];
+    }
+    const task = {
+      index: tasks.length,
+      descripton: userTask,
+      completed: false,
+    };
+    tasks.push(task);
+  } else {
+    isEditTask = false;
+    tasks[editId].descripton = userTask;
+  }
+  taskInput.value = '';
+  localStorage.setItem('task-list', JSON.stringify(tasks));
+  addElemToPage();
+});
+
+taskInput.addEventListener('keyup', (e) => {
+  const userTask = taskInput.value;
+  if (e.key === 'Enter' && userTask) {
+    if (!isEditTask) {
+      if (!tasks) {
+        tasks = [];
+      }
+      const task = {
+        index: tasks.length,
+        descripton: userTask,
+        completed: false,
+      };
+      tasks.push(task);
+    } else {
+      isEditTask = false;
+      tasks[editId].descripton = userTask;
+    }
+    taskInput.value = '';
+    localStorage.setItem('task-list', JSON.stringify(tasks));
+    addElemToPage();
+  }
+});
+
+window.del = (deletId) => {
+  tasks.splice(deletId, 1);
+  localStorage.setItem('task-list', JSON.stringify(tasks));
```

Comment:
[OPTIONAL] Deleting tasks does not work as expected. I created three tasks, deleted the first, and when I went to delete the second, the third task was deleted instead of the second. I was unable to delete the second task when I tried to. I recommend fixing that so that your project is complete.